arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -265,24 +265,10 @@ static void up_dumpstate(void)
 
   if (sp > istackbase - istacksize && sp < istackbase)
     {
-      uint32_t *stackbase;
```

Comment:
should we remove -4 at line 255?

Reply:
@xiaoxiang781216 
I think so too, becuase in arm_syscall.c we can find the following code.

```
#ifdef CONFIG_ARCH_KERNEL_STACK                                                                                                                                                                                                       
          /* If this is the first SYSCALL and if there is an allocated                                                                                                                                                                
           * kernel stack, then switch to the kernel stack.                                                                                                                                                                           
           */                                                                                                                                                                                                                         
                                                                                                                                                                                                                                      
          if (index == 0 && rtcb->xcp.kstack != NULL)                                                                                                                                                                                 
            {                                                                                                                                                                                                                         
              rtcb->xcp.ustkptr = (FAR uint32_t *)regs[REG_SP];                                                                                                                                                                       
              regs[REG_SP]      = (uint32_t)rtcb->xcp.kstack +                                                                                                                                                                        
                                  ARCH_KERNEL_STACKSIZE;                                                                                                                                                                              
            }                                                                                                                                                                                                                         
#endif               
```

However, if we remove -4 at line 255, I think I should add a separate commit to this PR,
instead of ammending exsisting commit. What do you think?

Reply:
Sure.

Reply:
Done

Reply:
Done